nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,449 @@
+    stat_map_base64 = encodebytes(stat_map_sprite.read()).decode('utf-8')
+    stat_map_sprite.close()
+
+    if colorbar:
+        stat_map_cm.seek(0)
+        cm_base64 = encodebytes(stat_map_cm.read()).decode('utf-8')
+        stat_map_cm.close()
+    else:
+        cm_base64 = ''
+
+    # Load the sprite meta-data from the json dump
+    stat_map_json.seek(0)
+    params = json.load(stat_map_json)
+    stat_map_json.close()
+
+    # Convet cut coordinates into cut slices
+    cut_slices = np.round(apply_affine(
+        np.linalg.inv(stat_map_img.affine), cut_coords))
+
+    # Create a json-like structure
+    # with all the brain sprite parameters
+    sprite_params = {
+                        'canvas': '3Dviewer',
+                        'sprite': 'spriteImg',
+                        'nbSlice': params['nbSlice'],
+                        'overlay': {
+                                'sprite': 'overlayImg',
+                                'nbSlice': params['nbSlice'],
+                                'opacity': opacity
+                                },
+                        'colorBackground': cbg,
+                        'colorFont': cfont,
+                        'crosshair': draw_cross,
+                        'affine': params['affine'],
+                        'flagCoordinates': annotate,
+                        'title': title,
+                        'flagValue': annotate,
+                        'numSlice': {
+                            'X': cut_slices[0],
+                            'Y': cut_slices[1],
+                            'Z': cut_slices[2]
+                        },
+                    }
+    if colorbar:
+        sprite_params['colorMap'] = {
+                    'img': 'colorMap',
+                    'min': params['min'],
+                    'max': params['max']
+                }
+
+    # Load javascript libraries
+    js_dir = os.path.join(os.path.dirname(__file__), 'data', 'js')
+    with open(os.path.join(js_dir, 'jquery.min.js')) as f:
+        js_jquery = f.read()
+    with open(os.path.join(js_dir, 'brainsprite.min.js')) as f:
+        js_brainsprite = f.read()
+
+    # Load the html template, and plug base64 data and meta-data
     html = get_html_template('stat_map_template.html')
-    html = html.replace(
-        'INSERT_STAT_MAP_DATA_HERE', '"{}"'.format(_encode_nii(stat_map_img)))
-    html = html.replace('INSERT_MNI_DATA_HERE', encoded_bg)
-    html = html.replace('INSERT_STAT_MAP_PARAMS_HERE',
-                        json.dumps(stat_map_params))
-    return StatMapView(html)
+    html = html.replace('INSERT_SPRITE_PARAMS_HERE', json.dumps(sprite_params))
+    html = html.replace('INSERT_BG_DATA_HERE', bg_base64)
+    html = html.replace('INSERT_STAT_MAP_DATA_HERE', stat_map_base64)
+    html = html.replace('INSERT_CM_DATA_HERE', cm_base64)
+    html = html.replace('INSERT_JQUERY_HERE', js_jquery)
+    html = html.replace('INSERT_BRAINSPRITE_HERE', js_brainsprite)
```

Comment:
this function is too long, please divide it in smaller ones

Reply:
was also brought up by @kchawla-pi . Will work on that.

Reply:
thanks!